src/tools/bit2core.c

```diff
@@ -60,6 +65,37 @@ rev_info parse_revision_arg(const char* rev)
   exit(-1);
 }
 
+rev_info* find_board_revision_from_targetname(char* targetname)
+{
+  int revcnt = sizeof(revs) / sizeof(rev_info);
+
+  for (int idx = 0; idx < revcnt; idx++)
+  {
+    if (strcmp(targetname, revs[idx].target_device_name) == 0)
+    {
+      printf("Bitstream's target device is detected as: \"%s\" (%s)\n", revs[idx].name, revs[idx].target_device_name);
+      return &revs[idx];
+    }
+  }
+
+  printf("This bitstream is for an unknown target device: \"???\" (%s)\"\n", targetname);
+  return NULL;
+}
+
+int check_bitstream_header_targetdevice(rev_info rev, unsigned char* bitstream)
```

Comment:
... then here you would check the selected target against the FPGA part detected, to determine if it can match (remembering that some FPGA parts might be used in multiple targets)

Reply:
Yep, that is exactly what is happening here, despite our vocabulary differences :)